aider_lint_fixer/pattern_matcher.py

```
@@ -997,34 +997,62 @@ def learn_from_fix(self, error_message: str, language: str, linter: str, fix_suc
             try:
-                with open(training_file, "r", encoding="utf-8") as f:
-                    training_data = json.load(f)
-            except Exception as e:
-                logger.warning(f"Failed to load training data for {language}: {e}")
+                # Read existing data
                 training_data = []
+                if training_file.exists():
+                    try:
+                        with open(training_file, "r", encoding="utf-8") as f:
+                            training_data = json.load(f)
+                    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)